sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/xml/XmlExpressionEvalUtils.scala

```diff
@@ -40,3 +41,98 @@ object XmlExpressionEvalUtils {
     UTF8String.fromString(dataType.sql)
   }
 }
+
+object XPathEvaluatorFactory {
+  def create(dataType: DataType, path: UTF8String): XPathEvaluator = {
+    dataType match {
+      case BooleanType => XPathBooleanEvaluator(path)
+      case ShortType => XPathShortEvaluator(path)
+      case IntegerType => XPathIntEvaluator(path)
+      case LongType => XPathLongEvaluator(path)
+      case FloatType => XPathFloatEvaluator(path)
+      case DoubleType => XPathDoubleEvaluator(path)
+      case dt if dt.isInstanceOf[StringType] => XPathStringEvaluator(path)
```

Comment:
nit: just for consistency w/ other places:
```suggestion
      case _: StringType => XPathStringEvaluator(path)
```